src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -36,7 +36,8 @@ export class HttpResourceFetcher implements ResourceFetcher {
         return new Promise<string>((resolve, reject) => {
             request(this.url, (error: any, response: IncomingMessage, body: any) => {
                 if (error) {
-                    reject(error)
+                    // Throw sanitized error so we don't end up revealing any potentially private URLs.
+                    reject(new Error('Error loading from URL'))
```

Comment:
Will fully obscuring the error make it harder to diagnose issues? The error may contain other attributes (like an error code) which calling code may want to react to. Is it feasible to obscuring just the url from the error message, but otherwise leave the error object as-is?

Reply:
I haven't done a deep dive on where the URL can surface in the error message. Feels kinda hacky to remove/alter fields from the request object but I can think about it. I'm not 100% sure what information we'd get from a failed request though.

Reply:
In a stunning turn of events, I made it possible to show the URL and also did a hacky thing. Score one for @awschristou .